### ISSUE TITLE
`reportIncompatibleUnannotatedOverride` false negative when base class attribute is assigned to a callable

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#1289]
### Description

Code sample in [basedpyright playground](https://basedpyright.com/?typeCheckingMode=all&reportUnannotatedClassAttribute=false&code=GYJw9gtgBAhgRgYygSwgBzCALlAggIQGEBYAKDIQBsYBnGqAMTDAAoDCBKALjKj6mDMoAXijUIcACYwuUAHJgAdgFMya0lVr18MECyZhuvfoLAioARiA)

```python
from abc import ABC

class Foo(ABC):
    foo = lambda: None


class Bar(Foo):
    foo = 1 # no error
```
